docs/_docs/02_features/debugging.md

```diff
@@ -0,0 +1,146 @@
+
+Terragrunt provides a debug mode you can access through the `--terragrunt-debug`
+command flag. For example you could use it like this to debug an `apply`
+that's producing unexpected output:
+
+    $ terragrunt --terragrunt-debug apply
+
+Running this command will do two things for you:
+  - Output a file named `terragrunt-debug.tfvars` to your terraform working
+    directory (the same one containing your `terragrunt.hcl` or `main.tf`
+    terraform code)
+  - Print instructions on how to invoke terraform against the generated file to
+    reproduce exactly the same terraform output as you saw when invoking
+    `terragrunt`. This will help you to understand whether the problem is coming
+    from
+
+The flag's goal is to help you determine which of these three major areas is the
+root cause of your problem:
+  1. Misconfiguration of your infrastructure code.
+  2. An error in `terragrunt`.
+  3. An error in `terraform`.
+  
+
+Let's run through a few use-cases.  
+
+### Use-case: I use locals or dependencies in terragrunt.hcl, and the terraform output isn't what I expected
+ 
+Consider this file structure for a fictional production environment where we
+have configured an application to deploy as many tasks as there are minimum
+number of machines in some cluster.
+
+```
+    └── live
+        └── prod
+            └── app
+            |   ├── vars.tf
+            |   ├── main.tf
+            |   ├── outputs.tf
+            |   └── terragrunt.hcl
+            └── ecs-cluster
+                └── outputs.tf
+
+```
+
+The files contain this text (`app/main.tf` and `ecs-cluster/outputs.tf` elided
+for brevity):
+
+```hcl
+// app/vars.tf
+variable "image_id" {
+  type = string
+}
+
+variable "num_tasks" {
+  type = number
+}
+
+
+// app/outputs.tf
+output "task_ids" {
+  value = ["${module.app_infra_module.task_ids}"]
+}
+
+
+// app/terragrunt.hcl
+locals {
+  image_id = "acme/myapp:1"
+}
+
+dependency "cluster" {
+  config_path = "../ecs-cluster"
+}
+
+inputs = {
+  image_id = locals.image_id
+  num_tasks = dependency.cluster.outputs.cluster_min_size
+}
+```
+
+You perform a `terragrunt apply`, and find that oops! `outputs.task_ids` has 7
+elements, but you know that the cluster only has 4 VMs in it! What's happening?
+Let's figure it out. Run this:
+
+    $ terragrunt --terragrunt-debug apply
+
+After applying, you will see this output on standard error
+
+```
+[terragrunt] Variables passed to terraform are located in "/Users/erem/live/prod/app/terragrunt-debug.tfvars"
+[terragrunt] Run this command to replicate how terraform was invoked:
+[terragrunt]     terraform apply -var-file="/Users/erem/live/prod/app/terragrunt-debug.tfvars" "/Users/erem/live/prod/app"
+```
+
+Well we may have to do all that, but first let's just take a look at `terragrunt-debug.tfvars`
+
+```hcl
+// terragrunt-debug.tfvars
+
+// File generated by terragrunt-debug on 01/01/2020 05:10:16.000Z
+// 
+// These variables were used to invoke terraform in debug mode.
```

Comment:
NIT: This sentence as read makes it seem like we are invoking terraform in debug mode.

Reply:
Fixed the language